app/src/main/java/org/mozilla/fenix/home/HomeMenu.kt

```diff
@@ -211,9 +249,155 @@ class HomeMenu(
+            primaryTextColor
+        ) {
+            onItemTapped.invoke(Item.Extensions)
+        }
+
+        val syncSignInItem = BrowserMenuImageText(
+            context.getString(R.string.library_synced_tabs),
+            R.drawable.ic_synced_tabs,
+            primaryTextColor
+        ) {
+            onItemTapped.invoke(Item.SyncTabs)
+        }
+
+        val requestDesktopSiteItem = BrowserMenuImageSwitch(
+            imageResource = R.drawable.ic_desktop,
+            label = context.getString(R.string.browser_menu_desktop_site),
+            initialState = { false }
+        ) { checked ->
+            onItemTapped.invoke(Item.RequestDesktopSite(checked))
+        }
+
+        val whatsNewItem = BrowserMenuHighlightableItem(
+            context.getString(R.string.browser_menu_whats_new),
+            R.drawable.ic_whats_new,
+            iconTintColorResource = primaryTextColor,
+            highlight = BrowserMenuHighlight.LowPriority(
+                notificationTint = getColor(context, R.color.whats_new_notification_color)
+            ),
+            isHighlighted = { WhatsNew.shouldHighlightWhatsNew(context) }
+        ) {
+            onItemTapped.invoke(Item.WhatsNew)
+        }
+
+        val helpItem = BrowserMenuImageText(
+            context.getString(R.string.browser_menu_help),
+            R.drawable.ic_help,
+            primaryTextColor
+        ) {
+            onItemTapped.invoke(Item.Help)
+        }
+
+        val settingsItem = BrowserMenuImageText(
+            context.getString(R.string.browser_menu_settings),
+            R.drawable.ic_settings,
+            primaryTextColor
+        ) {
+            onItemTapped.invoke(Item.Settings)
+        }
+
+        // Only query account manager if it has been initialized.
+        // We don't want to cause its initialization just for this check.
+        val accountAuthItem =
+            if (context.components.backgroundServices.accountManagerAvailableQueue.isReady()
+                && context.components.backgroundServices.accountManager.accountNeedsReauth()) {
+                reconnectToSyncItem
+            } else {
+                null
+            }
+
+        val menuItems = listOfNotNull(
+            if (shouldUseBottomToolbar) null else menuToolbar,
+            if (settings.shouldDeleteBrowsingDataOnQuit) quitItem else null,
+            bookmarksItem,
+            historyItem,
+            downloadsItem,
+            extensionsItem,
+            syncSignInItem,
+            accountAuthItem,
+            BrowserMenuDivider(),
+            requestDesktopSiteItem,
+            BrowserMenuDivider(),
+            whatsNewItem,
+            helpItem,
+            settingsItem,
+            if (shouldUseBottomToolbar) BrowserMenuDivider() else null,
+            if (shouldUseBottomToolbar) menuToolbar else null
+        ).also { items ->
+            items.getHighlight()?.let { onHighlightPresent(it) }
+        }
```

Comment:
It made sense to me when we separated out to old vs new MenuItems in the `DefaultToolbarMenu` before because the menu items appearance were different, but since this is only reordering and I am assuming we aren't changing the icons. Should we just keep one instance of this initialization function and just have an if statement block for the different ordering to avoid duplicating all the menu item variables.

Reply:
We're reordering **and** adding the navigation and request desktop item. 

I think my approach will be easier when we are going back and deleting the old items: we won't need to decipher any of the logic, just remove the whole block of `oldCoreItems`.